.github/workflows/build-docker-image.yml

```diff
@@ -1,86 +1,75 @@
-        id: get-matrix
-        run: |
-          matrix=$(cat ci/shared/matrix/github.json | jq '.matrix' | jq -c '[ .[] | select( .image != null) ]')
-          echo "runner=$(echo $matrix)" >> $GITHUB_OUTPUT
     outputs:
-      matrix: ${{ steps.get-matrix.outputs.runner }}
+      targets: ${{ steps.list-targets.outputs.targets }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
 
-  build-image:
-    name: Build ${{ matrix.runner.name }}
-    runs-on: ${{ matrix.runner.runs-on }}
-    needs: get-matrix
+      - name: List targets
+        id: list-targets
+        uses: docker/bake-action/subaction/list-targets@v6
+        with:
+          target: build-all
+
+  bake:
+    runs-on: ubuntu-latest
+    needs:
+      - prepare
     strategy:
       fail-fast: false
       matrix:
-        runner: ${{ fromJson(needs.get-matrix.outputs.matrix ) }}
-    steps:
-      - uses: actions/checkout@v4
+        target: ${{ fromJson(needs.prepare.outputs.targets) }}
 
-      - uses: actions/setup-python@v5
+    steps:
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
         with:
-          python-version: "3.12"
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
 
-      - run: pip install pyyaml
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
 
-      - name: Build Docker image
-        run: |
-          python ci/build-docker-image.py ${{ matrix.runner.image }}
-          docker image inspect ${{ matrix.runner.image }}
+      - name: Build
+        uses: docker/bake-action@v6
+        with:
+          targets: ${{ matrix.target }}
+          push: false
+          load: true
 
-      - name: Test Docker image (VT build & test)
+      - name: Test (vt build)
         run: |
-          CMD='echo "CC=$CC" ; \
-          echo "CXX=$CXX" ; \
-          echo "FC=$FC" ; \
-          echo "CMPLR_ROOT=$CMPLR_ROOT" ; \
-          echo "CMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH" ; \
-          echo "CPATH=$CPATH" ; \
-          echo "INFOPATH=$INFOPATH" ; \
-          echo "INTEL_LICENSE_FILE=$INTEL_LICENSE_FILE" ; \
-          echo "LIBRARY_PATH=$LIBRARY_PATH" ; \
-          echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH" ; \
-          echo "ONEAPI_ROOT=$ONEAPI_ROOT" ; \
-          echo "PATH=$PATH" ; \
-          echo "TBBROOT=$TBBROOT" ; \
-          mkdir -p "/opt/vt/src" "/opt/vt/build/vt" ; \
-            git clone https://github.com/DARMA-tasking/vt /opt/vt/src ; \
-            cd /opt/vt/src ; \
-            bash ci/build_cpp.sh /opt/vt/src /opt/vt/build ; \
-            bash ci/test_cpp.sh /opt/vt/src /opt/vt/build ; \
-            bash ci/build_vt_sample.sh /opt/vt/src /opt/vt/build ;
-            rm -rf "/opt/vt/src" "/opt/vt/build"'
-          echo "Running ${CMD}"
+          TAG=${{ matrix.target }}
+          TAG=${TAG/-04/.04}
+          TAG=${TAG/3-16/3.16}
+          TAG=${TAG/11-2-2/11.2.2}
+          TAG=${TAG/12-2-0/12.2.0}
```

Comment:
This converts the target name into a tag.
In the .hcl we have to replace `.` with `-` in target name to appease bake. We bring the `.` back here in this somewhat ugly way :shrug: